### PR TITLE
Update filters.py

### PR DESCRIPTION
**Pull request by surajiyer**
We get a KeyError issue in feature_selection/filters.py L197-203 when certain (treatment, y_label) combinations do not occur naturally in the dataset, e.g., if we never see label=1 for "treatment1".

## Proposed changes

We can smooth label counts per treatment in _GetNodeSummary() by adding 1 for (treatment, label) combinations that do not occur naturally in the data. Issue opened @ https://github.com/uber/causalml/issues/280

## Types of changes

What types of changes does your code introduce to **CausalML**?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/uber/causalml/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/uber/causalml)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc. This PR template is adopted from [appium](https://github.com/appium/appium).

### CODE REVIEW

File: causalml/feature_selection/filters.py [hunk 1]
```diff
@@ -191,7 +198,12 @@ def _GetNodeSummary(data,
         for ti in treatment_group_keys: 
             results.update({ti: {}}) 
             for ci in y_name_keys:
-                results[ti].update({ci: results_series[ti, ci]}) 
+                if smooth:
+                    results[ti].update({ci: results[ti].update({ci: results_series[ti, ci]
```
Comment (paullo0106) [inline causalml/feature_selection/filters.py:202]: Seems `ci` will all be `None` as dictionary updating method doesn't return any value, I don't think that's what you are trying to do here?

File: causalml/inference/meta/tlearner.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
 from scipy.stats import norm
 from sklearn.exceptions import ConvergenceWarning
 from sklearn.neural_network import MLPRegressor
-from sklearn.utils.testing import ignore_warnings
+from sklearn.utils._testing import ignore_warnings
```
Comment (paullo0106) [inline causalml/inference/meta/tlearner.py:8]: let's use the latest version handling changes in #283

File: causalml/inference/tree/models.py [hunk 3]
```diff
@@ -18,7 +18,7 @@
 import numpy as np
 import scipy.stats as stats
 import pandas as pd
-from sklearn.utils.testing import ignore_warnings
+from sklearn.utils._testing import ignore_warnings
```
Comment (paullo0106) [inline causalml/inference/tree/models.py:21]: same as above

**Review by paullo0106 — CHANGES_REQUESTED** [review]
Thanks for contributing! Please find my comments inline

File: causalml/feature_selection/filters.py (hunk 1 above)
Comment (surajiyer) [reply causalml/feature_selection/filters.py]: Fixed. That was a clumsy copy-paste job that I overlooked somehow.

File: causalml/inference/meta/tlearner.py (hunk 2 above)
Comment (surajiyer) [reply causalml/inference/meta/tlearner.py]: updated with changes from #283

File: causalml/inference/tree/models.py (hunk 3 above)
Comment (surajiyer) [reply causalml/inference/tree/models.py]: updated from changes from #283

**Comment by surajiyer** [comment]
Currently the build is failing because shap is missing **_IPython_** as dependency. There is a PR related to fix this issue with Shap https://github.com/slundberg/shap/pull/1749

**Review by paullo0106 — APPROVED** [review]
Thanks for the contribution!
